File: Session13/Java/MergeKSortedArrays.java
https://practice.geeksforgeeks.org/problems/merge-k-sorted-arrays/1

class Pair
{
    int val;
    int x;
    int y;
    Pair(int val,int x, int y)
    {
        this.val=val;
        this.x=x;
        this.y=y;
    }
}

class Solution
{
    //Function to merge k sorted arrays.
    public static ArrayList<Integer> mergeKArrays(int[][] arr,int K) 
    {
        PriorityQueue<Pair> minHeap= new PriorityQueue<>((a,b) -> a.val-b.val);
        ArrayList<Integer> ans = new ArrayList<>();
        for(int i=0;i<arr.length;i++)
        {
            minHeap.add(new Pair(arr[i][0], i, 0));
        }

        while(!minHeap.isEmpty())
        {
            int x=minHeap.peek().x;
            int y=minHeap.peek().y;
            ans.add(minHeap.poll().val);

            if(y+1<K)
            {
                minHeap.add(new Pair(arr[x][y+1], x, y+1));
            }
        }
        return ans;
    }
}

File: Session13/C++/MergeKSortedArrays.cpp
https://practice.geeksforgeeks.org/problems/merge-k-sorted-arrays/1

typedef pair<int, pair<int, int>> node;
class Solution
{
    public:
    //Function to merge k sorted arrays.
    vector<int> mergeKArrays(vector<vector<int>> arr, int K)
    {
        //code here
        priority_queue<node, vector<node>, greater<node>> pq;
        for(int i=0; i<K; i++)
        {
            pq.push({arr[i][0], {i, 0}});
        }
        
        vector<int> res;
        while(!pq.empty())
        {
            node current = pq.top();
            pq.pop();
            res.push_back(current.first);
            
            int x = current.second.first;
            int y = current.second.second;
            
            if(y+1 < K)
            {
                pq.push({arr[x][y+1], {x, y+1}});
            }
        }
        return res;
    }
};